Core/Main/itkTransformixFilter.hxx

```diff
@@ -237,6 +237,14 @@ TransformixFilter<TImage>::GenerateData()
     }
   }
 
+  if (m_ExternalTransform)
+  {
+    auto & transformParameterMap = transformParameterMapVector.front();
+    SetParameterValueAndWarnOnOverride(transformParameterMap, "Transform", "ExternalTransform");
+    SetParameterValueAndWarnOnOverride(
+      transformParameterMap, "TransformAddress", elx::Conversion::ObjectPtrToString(m_ExternalTransform));
```

Comment:
Note that for TransformixFilter, a specified external ITK transform does not _replace_ the parameter maps. Instead, its "TransformAddress" is just added to the first transform parameter map.

Transform parameter maps may have extra information that are not represented by an external ITK transform, specifically the "ResampleInterpolator" and the output image domain specification (Size/Spacing/Origin/Index/Direction)

Reply:
ok, but does it replace the previous transform information? Here it should not act as an initial transform, just as the main transform

Reply:
> does it replace the previous transform information? 

@mstaring When the user has provided a _single_ transform parameter map, it does. When the user has provided a vector of _multiple_ transform parameter maps, it only replaces the transform information of the first map. Is that OK?

In either case, it will always replace the transform specified by a previous call to the old `SetTransform` member function.

Reply:
Then it would act as the initial transform of the other transformations. The expected behaviour for transformix is that the transform that you set, is the one that is evaluated. So, that would mean deleting the other transforms in the vector of parameter maps, and only keep the first (or last?) one and replace it with the ExternalTransform

Reply:
> that would mean deleting the other transforms in the vector of parameter maps, and only keep the first (or last?) one and replace it with the ExternalTransform

Maybe then it would be best to only keep the last map from the transform parameter maps. Because that's the one that should contain the "ResampleInterpolator" and the output image domain specification (Size/Spacing/Origin/Index/Direction), right?

Another option could be to require exactly 1 map. And throw an exception when the user-specified vector of transform parameter maps has more than 1 item. What do you think?